### PR TITLE
Copter: remove pv_ functions that duplicate Location functions

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

File: ArduCopter/GCS_Mavlink.cpp [hunk 1]
```diff
@@ -1244,14 +1244,11 @@ void GCS_MAVLINK_Copter::handleMessage(mavlink_message_t* msg)
                 case MAV_FRAME_GLOBAL:
                 case MAV_FRAME_GLOBAL_INT:
                 default:
-                    // pv_location_to_vector does not support absolute altitudes.
-                    // Convert the absolute altitude to a home-relative altitude before calling pv_location_to_vector
-                    loc.alt -= copter.ahrs.get_home().alt;
-                    loc.relative_alt = true;
```
Comment (khancyr) [inline ArduCopter/GCS_Mavlink.cpp]: you need to keep this one otherwise alt will be in ALT_FRAME_ABSOLUTE and not ALT_FRAME_ABOVE_HOME

Comment (peterbarker) [reply ArduCopter/GCS_Mavlink.cpp]: Why is that a problem?  the call on `Location loc` will cope.

**Comment by rmackay9** [comment]
We just need to be sure that we are correctly handling the cases where the Location functions don't return a value.  So previously I think it would end up using a stale position while now it will use do something else.. sorry to be lazy but what is the "something else"?

It looks like maybe for the ROI yaw it will stop doing ROI permanently?  I.e. it was pointing the nose/gimbal at the point but from then on it will stop.  This may cause an undesired change in behaviour in the unlikely case that the DO_ROI command is the first in a mission and the vehicle is switched into Auto while still disarmed and before it has a good position estimate.  In think in master the ROI would start working when the position estimate became good while with this change it will never ROI.

**Comment by peterbarker** [comment]
On Tue, 5 Feb 2019, Randy Mackay wrote:

> It looks like maybe for the ROI yaw it will stop doing ROI permanently? I.e.
> it was pointing the nose/gimbal at the point but from then on it will stop.
> This may cause an undesired change in behaviour in the unlikely case that
> the DO_ROI command is the first in a mission and the vehicle is switched
> into Auto while still disarmed and before it has a good position estimate.
> In think in master the ROI would start working when the position estimate
> became good while with this change it will never ROI.

The roi itself will be invalid, and nothing is going to update the roi 
after this point.  The roi is a Vector3f in the AutoYaw class.  If you 
wanted to have *any* chance of getting a valid vector later we'd have to 
store the lat/lon in there and repeatedly try making a vector out of it 
until we succeed.

I think it *is* a mistake to change the yaw mode on failure, 'though - 
I'll remove that.

**Comment by rmackay9** [comment]
@peterbarker, sounds good, txs

**Comment by peterbarker** [comment]
@rmackay9 Done and pushed in.

**Comment by rmackay9** [comment]
Merged, thanks!

**Pull request merged**